src/InvestJob.jsx

```diff
@@ -143,30 +129,32 @@ export class InvestJob extends React.Component {
       {sessionID: value});
   }
 
-  loadState(sessionFilename) {
+  async loadState(sessionFilename) {
     /** Set this component's state to the object parsed from a JSON file.
     *
     * @params {string} sessionFilename - path to a JSON file.
     */
 
-    // const filename = path.join(CACHE_DIR, sessionFilename);
     if (fs.existsSync(sessionFilename)) {
       const loadedState = JSON.parse(fs.readFileSync(sessionFilename, 'utf8'));
+      
+      // Right now we saveState is only called w/in investExecute and only
+      // after invest has created a logfile, which means an invest logfile
+      // should always exist and can be used to get args values and initialize SetupTab.
+      const datastack = await fetchDatastackFromFile(
+        { datastack_path: loadedState.logfile })
+      loadedState['argsInitDict'] = datastack['args']
+      Object.assign(loadedState, {setupKey: changeSetupKey(this.state.setupKey)})

```

Comment:
Okay, so changing this artificial `key` will cause `SetupTab` component to create a new component instance rather than update an old one. This will update all the values... Something like that?

Reply:
Yes! Creating the new component basically means calling SetupTab's `constructor` and `componentDidMount()`. That initializes some state and does the sorting and ordering of args. Basically all the stuff that needs to happen when a new model/ARGS_SPEC is selected, but not everytime the SetupTab renders (i.e. as user's interact with the form).

Reply:
It does seem like a bit of hacky way of controlling code execution, but it's one of the recommended patterns: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key